zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -738,11 +1121,29 @@ pub(crate) fn put_transparent_output<P: consensus::Parameters>(
     txout: &TxOut,
     output_height: Option<BlockHeight>,
     address: &TransparentAddress,
-    receiving_account_uuid: AccountUuid,
     known_unspent: bool,
-) -> Result<UtxoId, SqliteClientError> {
+) -> Result<(AccountRef, KeyScope, UtxoId), SqliteClientError> {
+    let addr_str = address.encode(params);
+
+    // Unlike the shielded pools, we only can receive transparent outputs on addresses for which we
+    // have an `addresses` table entry, so we can just query for that here.
+    let (address_id, account_id, key_scope_code) = conn.query_row(
+        "SELECT id, account_id, key_scope
+         FROM addresses
+         WHERE cached_transparent_receiver_address = :transparent_address",
+        named_params! {":transparent_address": addr_str},
+        |row| {
+            Ok((
+                row.get("id").map(AddressRef)?,
+                row.get("account_id").map(AccountRef)?,
+                row.get("key_scope")?,
+            ))
+        },
+    )?;
```

Comment:
It would be nice to still return `AddressNotRecognised` like the caller `put_received_transparent_utxo` did previously.